app/controllers/homebuyings_controller.rb

```diff
@@ -42,40 +42,49 @@ def create
   end
 
   def show
-    @homebuying = Homebuying.find(params[:id])
+    @homebuying = Homebuying.find(params[:id]) if current_client
   end
 
   def edit
-    if current_client
-      @homebuying = current_client.homebuying
-    elsif current_user
-      @foreclosure = Homebuying.find(params[:id])
-    end
+    @homebuying = current_client.homebuying if current_client
+    # if current_client
```

Comment:
Also, if we're going to use `current_client` instead of `params[:id]` should we do some sort of `authenticate` before action? What happens if an unsigned client tries to access this action?